Translation Add React to a website (#72)
reactjs/fr.reactjs.org: content/docs/add-react-to-a-website.md

---
id: add-react-to-a-website

> title: Ajouter React à votre site web

Author: Peut etre plus impersonnel, mais après tout pourquoi pas.
"Ajouter React à un site web" peut être ?

permalink: docs/add-react-to-a-website.html
redirect_from:
  - "docs/add-react-to-an-existing-app.html"
prev: getting-started.html
next: create-a-new-react-app.html
---

Utiliser React en fonction de vos besoins

> Depuis sa création, React s'adapte en fonction de vos besoins, ainsi vous pouvez l'utiliser autant que vous le souhaitez, ou simplement ajouter un peu d'interactivité à une page existante. Les composants React sont une bonne solution pour faire ça.

Reviewer: ```suggestion
React a été conçu dès le départ pour une adoption progressive, et **vous pouvez utiliser React _a minima_ ou autant que nécessaire.** Peut-être souhaitez-vous simplement ajouter un peu d'interactivité à une page existante. Les composants React sont une bonne solution pour ça.
```

Harmonisation avec le début identique dans *Bien démarrer*, et allègement mineur.

> La majorité des sites web ne sont pas, et n'ont pas besoin d'être, des applications web monopage (en anglais Single Page Application ou plus simplement SPA, NdT). Sans outils ou configurations supplémentaires, mais avec simplement quelques lignes de code, vous pouvez essayer React dans une petite partie de votre site web. Vous pourrez par la suite ajouter des composants supplémentaires ou garder React dans quelques éléments dynamiques.

Reviewer: ```suggestion
La majorité des sites web ne sont pas, et n'ont pas besoin d'être, des applis web monopages *(Single-Page Apps ou plus simplement SPA, NdT)*. **Avec juste quelques lignes de code et sans outillage particulier**, vous pouvez essayer React sur une petite partie de votre site web. Vous pourrez par la suite ajouter des composants supplémentaires ou limiter React à quelques éléments dynamiques.
```

- Formatage adapté de la NdT
- Retour du formatage VO, qui n'a rien d'anecdotique
- Fluidification

Author: Je suis pas très partisan de "applis" ou meme de "apps". Application est quand meme le terme officiel en anglais et en français. Du coup je serais plus partant pour garder ta correction mais avec "application".

```suggestion
La majorité des sites web ne sont pas, et n'ont pas besoin d'être, des applications web monopages *(Single-Page Application ou plus simplement SPA, NdT)*. **Avec juste quelques lignes de code et sans outillage particulier**, vous pouvez essayer React sur une petite partie de votre site web. Vous pourrez par la suite ajouter des composants supplémentaires ou limiter React à quelques éléments dynamiques.
```

Reviewer: @Nab69 le mot "appli" n'existe pas plus que "démo", et pourtant ils sont en usage courant, y compris à l'écrit, dans des documents officiels, etc. En pratique, la VO utilise parfois “application” (qu'on traduira tel quel), parfois “app” (qu'on traduira en “appli”), pour préserver le rythme et le ton de la VO.

Author: @tdd je suis pas non plus un grand fan de "démo" mais au moins celui-ci apparait dans des dictionnaires en ligne.
C'est peut être l'occasion de faire évoluer une mauvaise traduction en utilisant les bons termes ?
D'ailleurs je pense @sheplu partage mon avis sur ce point.

Reviewer: Tu imagines bien que les dicos en ligne ne pourront jamais suivre la réalité du terrain technologique. Et je ne suis pas sûr de l'intention derrière les termes « mauvaise traduction ».

Quoi qu'il en soit, je ne vais pas non plus me battre pour ce qui est, *in fine*, un détail qui ne corrompt pas le sens du texte, mais en altère simplement le rythme. Je t'invite donc à valider ta suggestion qui reprenait mon texte à ce terme près.

Author: Il n'y a pas de mauvaises intentions derrière ça, juste des traductions 😉 
Par ailleurs on peut quand même en débattre.

Reviewer: Je pense qu'on est arrivé au bout de ce débat ; prends ta dernière suggestion qui retient mon texte à « application » près, ça ira bien.

> La majorité des sites web ne sont pas, et n'ont pas besoin d'être, des applications web monopage (en anglais Single Page Application ou plus simplement SPA, NdT). Sans outils ou même de configurations supplémentaires, mais avec simplement quelques lignes de code, vous pouvez essayer React dans une petite partie de votre site web. Vous pourrez par la suite ajouter des composants supplémentaires ou garder React dans quelques éléments dynamiques.

Author: ouais ca me va mieux aussi ;)

---

- [Ajouter React en une minute](#add-react-in-one-minute)

> - [Optionnel : Essayer React avec JSX](#optional-try-react-with-jsx) (aucun paquet nécessaire!)

Reviewer: ```suggestion
- [Optionnel : Essayer React avec JSX](#optional-try-react-with-jsx) (aucun _bundler_ nécessaire !)
```

- Insécables
- Un _bundler_ n'est pas un paquet, mais un _empaqueteur_. Le terme VF est toutefois rare, on reste donc sur _bundler_ mais en italiques.

Author: oui empaqueteur ne plaisait pas donc j'ai cherché à passer par plus simple

## Ajouter React en une minute {#add-react-in-one-minute}

> Dans cette partie, nous allons montrer comment ajouter un composant React à une page HTML existante. Vous pouvez suivre les étapes avec votre propre site web ou avec une page HTML vide pour vous entrainer.

Reviewer: ```suggestion
Dans cette partie, nous allons voir comment ajouter un composant React à une page HTML existante. Vous pouvez suivre les étapes avec votre propre site web, ou avec une page HTML vide pour vous entrainer.
```

Il est plus agréable de se placer du point de vue apprenant (voir) qu'enseignant (montrer).

Author: ok, pourquoi pas en effet.

> Il n'y a pas besoin d'outils compliqués ou d'installations requises, **pour suivre cette section, vous avez juste besoin d'une connexion à internet et d'une minute de votre temps.**

Reviewer: ```suggestion
Il n'y a pas besoin d'outils compliqués ou d'installations à faire : **pour suivre cette section, vous avez juste besoin d'une connexion à internet et d'une minute de votre temps.**
```

OK pour retirer l'incise, mais on peut ici la transformer en deux-points, la virgule jouant plus un rôle de respiration dans une séquence.

Optionnel : [Télécharger l'exemple complet (2KB zippé)](https://gist.github.com/gaearon/6668a1f6986742109c00a581ce704605/archive/f6c882b6ae18bde42dcf6fdb751aae93495a2275.zip)

> ### Étape 1 : Ajouter un conteneur DOM à votre HTML {#step-1-add-a-dom-container-to-the-html}

Reviewer: ```suggestion
### Étape 1 : ajouter un conteneur DOM à votre HTML {#step-1-add-a-dom-container-to-the-html}
```

Pas de majuscule derrière le deux-points en français :wink:

> Premièrement, ouvrez la page HTML que vous souhaitez éditer. Ajouter une balise `<div>` vide afin de définir ce que vous souhaitez afficher avec React. Par exemple :

Reviewer: ```suggestion
Pour commencer, ouvrez la page HTML que vous souhaitez modifier. Ajoutez une balise `<div>` vide pour indiquer l'endroit dans lequel vous souhaitez afficher quelque chose avec React. Par exemple :
```

- « Premièrement » est très formel, et se décline mal par la suite.
- « Éditer » n'a pas le même sens en français
- Souci d'infinitif / impératif
- Gros glissement sémantique sur la fin

Author: Je ne trouve pas forcément que "premièrement" soit si différent de "pour commencer", mais ok 😄 
En effet c'est un anglicisme très utilisé mais modifier me va mieux.
Pour à peu près la même traduction pour la fin mais je la trouvais un peu lourde.

```html{3}
<!-- ... code HTML existant ... -->

<div id="like_button_container"></div>

<!-- ... code HTML existant ... -->
```

L'attribut `id` de votre `<div>`, qui est unique, va vous permettre de manipuler cette balise depuis le code JavaScript afin d'afficher un composant React dans celle-ci.

>Astuce
>

> >Vous pouvez placer une `<div>` qui sert de « conteneur », où vous le souhaitez dans la balise `<body>`. Vous pouvez, par ailleurs, avoir autant de conteneurs DOM indépendants que vous le souhaitez dans la même page. React remplacera le contenu existant de ces conteneurs, ils sont donc la plupart du temps vides.

Reviewer: ```suggestion
>Vous pouvez placer une telle `<div>` « conteneur » **où vous le souhaitez** dans la balise `<body>`. Vous pouvez d’ailleurs avoir autant de conteneurs DOM indépendants que vous le souhaitez dans la même page. Il sont généralement vides, car React remplacera leur contenu existant de toutes façons.
```

Légère reformulation (je pense que parler du côté vide en premier est plus impactant), et retour du formatage. C’est marrant ça, tu es le premier que je vois à décider de virer les emphases ?!

> ### Étape 2 : Ajouter les balises « Script » {#step-2-add-the-script-tags}

Reviewer: ```suggestion
### Étape 2 : ajouter les balises de script {#step-2-add-the-script-tags}
```

> Ensuite, ajouter trois balises `<script>` à votre page HTML juste avant la fermeture de la balise `</body>` :

Reviewer: ```suggestion
Ensuite, ajoutez trois balises `<script>` à votre page HTML, juste avant la fermeture de la balise `</body>` :
```

Tu as un vrai souci avec l'impératif dis donc :wink:

```html{5,6,9}
  <!-- ... autres contenus HTML ... -->

>   <!-- Chargez React. -->

Reviewer: ```suggestion
  <!-- Charger React. -->
```

Là pour le coup c'est descriptif, donc un infinitif :grin:

Author: En fait en relisant je vois bien le problème mais je trouve que même l'infinitif ca n'est pas tip top. Avec simplement "charge", je trouve ca plus compréhensible (ça se lit mieux non ?)

```suggestion
  <!-- Charge React. -->
```

Reviewer: Dans ce contexte précis, la 3e personne du présent de l'indicatif est en effet une bonne idée 😁

  <!-- Remarque : pour le déploiement, remplacez "development.js" par "production.min.js". -->
  <script src="https://unpkg.com/react@16/umd/react.development.js" crossorigin></script>
  <script src="https://unpkg.com/react-dom@16/umd/react-dom.development.js" crossorigin></script>

>   <!-- Chargez votre composant React. -->

Reviewer: ```suggestion
  <!-- Charger notre composant React. -->
```

Infinitif de description, et pronom impliquant d'origine

Author: Et je reprends le "charge" ici aussi

```suggestion
  <!-- Charges notre composant React. -->
```

  <script src="like_button.js"></script>

</body>
```

Les deux premières balises permettent de charger React, alors que la troisième charge le code de votre composant.

> ### Étape 3 : Créer un composant React {#step-3-create-a-react-component}

Reviewer: ```suggestion
### Étape 3 : créer un composant React {#step-3-create-a-react-component}
```

Créez un fichier nommé `like_button.js` dans le même dossier que votre page HTML.

Ouvrez et copiez **[le code de démarrage](https://cdn.rawgit.com/gaearon/0b180827c190fe4fd98b4c7f570ea4a8/raw/b9157ce933c79a4559d2aa9ff3372668cce48de7/LikeButton.js)** et collez son contenu dans votre fichier.

>Astuce
>

> >Ce code définit un composant appelé `LikeButton`. Ne vous en faites pas si vous ne comprenez pas son code tout de suite, nous aborderons tous les éléments de React un par un dans la suite de notre [tutoriel pratique](/tutorial/tutorial.html) et dans le [guide des principaux concepts](/docs/hello-world.html). Pour le moment, essayons simplement de l'afficher à l’écran !

Reviewer: ```suggestion
>Ce code définit un composant appelé `LikeButton`. Ne vous en faites pas si vous ne comprenez pas son code tout de suite, nous aborderons les facettes de React ultérieurement dans notre [tutoriel pratique](/tutorial/tutorial.html) et dans le [guide des fondamentaux](/docs/hello-world.html). Pour le moment, essayons simplement de l'afficher à l’écran !
```

- Bon choix de migration incise -> virgule :+1:
- En revanche, glissement sémantique sur la référence au tuto (on n'est pas encore dedans).

> Ajouter deux lignes à la fin de `like_button.js`, après **[le code de démarrage](https://cdn.rawgit.com/gaearon/0b180827c190fe4fd98b4c7f570ea4a8/raw/b9157ce933c79a4559d2aa9ff3372668cce48de7/LikeButton.js)** :

Reviewer: ```suggestion
Ajoutez deux lignes à la fin de `like_button.js`, après **[le code de démarrage](https://cdn.rawgit.com/gaearon/0b180827c190fe4fd98b4c7f570ea4a8/raw/b9157ce933c79a4559d2aa9ff3372668cce48de7/LikeButton.js)** :
```

```js{3,4}

> // ... le code de démarrage collé ...

Reviewer: ```suggestion
// ... le code de démarrage que vous avez collé ...
```

const domContainer = document.querySelector('#like_button_container');
ReactDOM.render(e(LikeButton), domContainer);
```

> Ces deux lignes de code vont remplacer le contenu de la `<div>` que nous avons ajoutée dans la première étape. Ainsi elles afficheront notre composant React sous forme de bouton « J’aime ».

Reviewer: Ce n'est pas vraiment le sens / l'articulation de la VO, mais bon, ça passe.

### Et voilà ! {#thats-it}

Il n'y a pas de quatrième étape. **Vous venez tout juste d'ajouter votre premier composant React à votre site web.**

> Pour plus d'astuces concernant l'intégration de React, n'hésitez pas à consulter les sections suivantes.

Reviewer: ```suggestion
Pour plus d'astuces concernant l'intégration de React, n'hésitez pas à consulter les prochaines sections.
```

« Suivant » est réservé au contenu qui suit effectivement dans la page courante. Pour des pages ultérieures, on utilisera plutôt « prochain », en position épithète.

**[Voir le code source intégral de l'exemple](https://gist.github.com/gaearon/6668a1f6986742109c00a581ce704605)**

> **[Télécharger le code source intégral de l'exemple (2KB zippé)](https://gist.github.com/gaearon/6668a1f6986742109c00a581ce704605/archive/f6c882b6ae18bde42dcf6fdb751aae93495a2275.zip)**

Reviewer: ```suggestion
**[Télécharger le code source intégral de l'exemple (2Ko zippé)](https://gist.github.com/gaearon/6668a1f6986742109c00a581ce704605/archive/f6c882b6ae18bde42dcf6fdb751aae93495a2275.zip)**
```

Unité française

> ### Astuce : Réutiliser un composant {#tip-reuse-a-component}

Reviewer: ```suggestion
### Astuce : réutiliser un composant {#tip-reuse-a-component}
```

Généralement, on souhaite afficher des composants React à plusieurs endroits d'une page HTML. Voici un exemple qui affiche le bouton « J’aime » à trois reprises et lui fournit quelques données :

[Voir le code source intégral de l'exemple](https://gist.github.com/gaearon/faa67b76a6c47adbab04f739cba7ceda)

> [Télécharger le code source intégral de l'exemple (2KB zippé)](https://gist.github.com/gaearon/faa67b76a6c47adbab04f739cba7ceda/archive/9d0dd0ee941fea05fd1357502e5aa348abb84c12.zip)

Reviewer: ```suggestion
[Télécharger le code source intégral de l'exemple (2Ko zippé)](https://gist.github.com/gaearon/faa67b76a6c47adbab04f739cba7ceda/archive/9d0dd0ee941fea05fd1357502e5aa348abb84c12.zip)
```

Unité française

>Remarque
>

> >Lorsque les éléments de React doivent être isolés les uns des autres, cette solution est souvent utilisée. Il est cependant plus facile d'utiliser le principe de [composition de composants](/docs/components-and-props.html#composing-components) à la place dans votre code React.

Reviewer: ```suggestion
>Lorsque les éléments de React doivent être isolés les uns des autres, cette solution est souvent utilisée. Il est cependant plus facile d'utiliser le principe de [composition de composants](/docs/components-and-props.html#composing-components) au sein de votre code React.
```

Ce paragraphe est déjà déroutant en VO ; je trouve qu'il rate sa cible, on ne comprend pas l'objectif (que j'ai bien perçu, mais qui est mal exprimé).  On fera une PR sur la VO d'abord, et on ajustera la VF suite à son traitement.

Author: Oui en effet, y'aura aussi plein d'autres endroits à uniformiser et modifier en VO

> ### Astuce : Minifier votre JavaScript en production {#tip-minify-javascript-for-production}

Reviewer: ```suggestion
### Astuce : minifier votre JavaScript pour la production {#tip-minify-javascript-for-production}
```

Micro-détail… « En » peut laisser penser que c'est dans l'environnement de production qu'on va réaliser la minification (genre `npm run build` dans une CD Heroku), ce qui n'est pas nécessairement le cas, ça peut être fait en amont pour préparer un livrable statique par exemple.

> ### Astuce : Minifié votre JavaScript en production {#tip-minify-javascript-for-production}

Author: ouais minifier me va pas mal

> Avant de déployer votre site web en production, il faut garder à l'esprit qu'un JavaScript non minifié risque de ralentir significativement le chargement de la page pour vos utilisateurs.

Reviewer: ```suggestion
Avant de déployer votre site web en production, gardez à l'esprit qu'un JavaScript non-minifié risque de ralentir significativement le chargement de la page pour vos utilisateurs.
```

On est sur de l'impératif, c'est plus impliquant qu'un « il faut » (ou pire, un « il convient de » bien désincarné…).

> Lorsque vous aurez minifié vos scripts d'application ainsi que référencé la version adaptée de React (soit `production.min.js`), vous serez alors prêt pour une mise en production :

Reviewer: ```suggestion
Si vous minifiez déjà vos scripts applicatifs, **votre site sera prêt pour la production** du moment que vous vous assurez de référencer, depuis le HTML déployé, les versions de React dont les noms se terminent en `production.min.js` :
```

Outre que ta forme est assez ampoulée, il y a un glissement sémantique sur le complément circonstanciel, qui n'est pas de temps mais de contexte.

Author: Désolé de te le dire, mais c'est ta version que je trouve trop "ampoulée". C'est pour cela que j'ai préféré modifier la phrase de la VO afin qu'elle soit plus compréhensible en français.
Je précise que j'ai d'ailleurs demandé quelques avis pour comparer les deux traductions, et ce même si je pense que la mienne n'est pas parfaite non plus.

Reviewer: @Nab69 mais ta version ne dit pas la même chose que la VO, elle en change le sens.

La VO dit que si on a déjà un système de minification en place, il nous suffit de basculer notre balise de chargement React vers leur bundle de prod pour avoir une application prod-ready.

Ta VF, outre des formes curieuses comme « ainsi que [participe passé] » ne distingue pas entre qqn qui a déjà la minification en place et les autres.

C'était là le sens de ma correction.

Author: Ok, je change en effet un peu le sens mais c'était pour ajouter de la simplicité dans la lecture. 
Je vois bien la volonté de garder le sens original dans la correction, mais en gardant une traduction plus proche de la VO (trop peut etre?), on perds en simplicité. Il faudrait un juste milieu mais je ne vois pas trop comment transformer ça (surtout après une grosse journée ^^)

Reviewer: Je comprends bien, mais j'ai du mal à souscrire à l'idée qu'on peut se permettre de changer volontairement le sens du texte original, on est là pour traduire, pas trahir (même si *traduttore, traditore*). Simplifier c'est une bonne chose, mais seulement pour alléger la formulation, pas au prix d'une perte d'info ou du nuance.

Author: J'ai bien compris aussi ta volonté de garder le sens et je suis d'accord, mais je pense qu'on pourrait trouver mieux.

Author: En retouchant un peu, qu'en penses-tu ?

```suggestion
Si vous minifiez déjà vos scripts applicatifs et si vous vous assurez de référencer les versions de React dont les noms se terminent en `production.min.js`, **votre site sera prêt pour la production** :
```

Reviewer: Ça me va !

Reviewer: Allez on finit tout ça que je puisse faire une repasse visuelle locale et fusionner dans `master`. Au passage ce serait cool si tu donnais les droits d'edit mainteneurs sur ta PR, que je puisse y retranscrire les éventuels tweaks de repasse visuelle (insécables, *soft hyphens*, typos, ce genre de choses)…

Author: @tdd Hum c'est bizarre parce que la case "allow edit from maintainers" est bien coché depuis le début... j'avais d'ailleurs laissé la valeur par défaut (qui est coché).

Je viens de la décocher/recocher pour forcer la màj mais c'est bizarre...

```js
<script src="https://unpkg.com/react@16/umd/react.production.min.js" crossorigin></script>
<script src="https://unpkg.com/react-dom@16/umd/react-dom.production.min.js" crossorigin></script>
```

> Si vous n'avez pas de marche à suivre pour la minification, voici [une façon de faire](https://gist.github.com/gaearon/42a2ffa41b8319948f9be4076286e1f3).

Reviewer: ```suggestion
Si vous n'avez pas de minification déjà en place pour vos scripts, voici [une façon de faire](https://gist.github.com/gaearon/42a2ffa41b8319948f9be4076286e1f3).
```

> ## Optionnel : Essayer React avec JSX {#optional-try-react-with-jsx}

Reviewer: ```suggestion
## Optionnel : essayer React avec JSX {#optional-try-react-with-jsx}
```

> Dans l'exemple ci-dessous, nous utilisons uniquement les fonctionnalitées natives supportées par le navigateur. C'est pourquoi nous faisons appel une fonction JavaScript pour indiquer à React ce qu'il faut afficher :

Reviewer: ```suggestion
Dans les exemples ci-dessus, nous utilisions uniquement les fonctionnalités prises en charge nativement par les navigateurs. C'est pourquoi nous appelions une fonction JavaScript pour indiquer à React ce qu'il fallait afficher :
```

Author: oups ah oui en effet sur ce coup là, je devrais être fatigué.

```js
const e = React.createElement;

// Affiche un bouton « J’aime »
return e(
  'button',
  { onClick: () => this.setState({ liked: true }) },
  'J’aime'
);
```

> Néanmoins, React nous offre aussi la possibilité d'utiliser [JSX](/docs/introducing-jsx.html) à la place.

Reviewer: ```suggestion
Néanmoins, React nous offre aussi la possibilité d'utiliser plutôt [JSX](/docs/introducing-jsx.html).
```

On bannit « à la place » (cf. Glossaire)

```js
// Affiche un bouton « J’aime »
return (
  <button onClick={() => this.setState({ liked: true })}>
    J’aime
  </button>
);
```

> Les deux extraits de code précédents sont en effet équivalent. Cependant même si la syntaxe **JSX est [complètement optionnelle](/docs/react-without-jsx.html)**, la plupart des développeurs React trouvent cela très pratique pour l'affichage (avec React ou même avec d'autres bibliothèques).

Reviewer: ```suggestion
Ces deux extraits de code sont équivalents. Même si la syntaxe **JSX est [complètement optionnelle](/docs/react-without-jsx.html)**, la plupart des gens la trouvent très pratique pour écrire du code  (avec React ou même avec d'autres bibliothèques).
```

Author: Je sais que j'ai mis "developpeurs React" à la place de "gens" car la VO est plûtot horrible sur ce point. C'est la première fois qu'on voit ce 'people' qui sort un peu de nulle part. Du coup je le garderais bien.

```suggestion
Ces deux extraits de code sont équivalents. Même si la syntaxe **JSX est [complètement optionnelle](/docs/react-without-jsx.html)**, la plupart des développeurs React la trouvent très pratique pour écrire du code  (avec React ou même avec d'autres bibliothèques).
```

Reviewer: @Nab69 j'avoue ne pas voir en quoi « gens » est « horrible ». C'est plus inclusif, tant sur le genre que sur le reste, que « développeurs React », tu ne trouves pas ?

Author: Disons que ca n'est pas déjà pas très valide dans le texte anglais car 'people' sort de nulle part. 
On peut toujours mettre développeur·euse·s par exemple ? (si je ne me trompe pas pour l'écriture inclusive)

```suggestion
Ces deux extraits de code sont équivalents. Même si la syntaxe **JSX est [complètement optionnelle](/docs/react-without-jsx.html)**, la plupart des développeur·euse·s React la trouvent très pratique pour écrire du code  (avec React ou même avec d'autres bibliothèques).
```

Reviewer: Le compromis est acceptable, même s'il est du coup un peu long dans la forme, prends ta dernière suggestion en date.

Vous pouvez tester JSX en utilisant un [convertisseur en ligne](http://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=Q&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&sourceType=module&lineWrap=true&presets=es2015%2Creact%2Cstage-2%2Cstage-3&prettier=true&targets=Node-6.12&version=6.26.0&envVersion=).

> ### Essayer sans tarder JSX {#quickly-try-jsx}

Author: oui, je trouvais ca sympa :smile:
je change du coup ?

> ### Essayer sans tarder JSX {#quickly-try-jsx}

Reviewer: ```suggestion
### Essayer JSX en un clin d’œil {#quickly-try-jsx}
```

La VO insiste sur la rapidité de mise en place, et en tous les cas ici l'adverbe / CCT sera mieux à la fin.

> ### Essayer sans tarder JSX {#quickly-try-jsx}

Author: c'est quand même très synonyme pour le coup 😄 et ca fait autant la promo comme le disait sheplu 😄

La façon la plus simple et rapide de tester JSX dans votre projet est d'ajouter la balise `<script>` ci-dessous à votre page :

```html
<script src="https://unpkg.com/babel-standalone@6/babel.min.js"></script>
```

> Vous pouvez dès à présent utiliser JSX simplement en ajoutant l'attribut `type="text/babel"` à n'importe quelle balise `<script>`. Vous pouvez télécharger et tester cet exemple contenant un [fichier HTML utilisant JSX](https://raw.githubusercontent.com/reactjs/reactjs.org/master/static/html/single-file-example.html).

Reviewer: ```suggestion
Vous pouvez désormais utiliser JSX dans les fichiers chargés par n'importe quelle balise `<script>` simplement en lui ajoutant l'attribut `type="text/babel"`. Vous pouvez télécharger et tester cet exemple contenant un [fichier HTML utilisant JSX](https://raw.githubusercontent.com/reactjs/reactjs.org/master/static/html/single-file-example.html).
```

Il faut garder l'association de la fonctionnalité à l'ajout de l'attribut ; ta VF pourrait laisser croire qu'ajouter l'attribut sur une des balises au hasard active la fonctionnalité pour le code chargé par toutes.

> Cette approche est pratique pour se former ou pour de simples démonstrations. Cependant, elle va ralentir l'affichage de votre site, **elle n'est donc pas à envisager en production**. Lorsque vous serez prêt à aller plus loin, supprimez la balise `<script>` et l'attribut `type="text/babel"` que vous venez d'ajouter. Dans la section suivante, nous verrons comment configurer le préprocesseur JSX afin de convertir automatiquement toutes les balises `<script>`.

Reviewer: ```suggestion
Cette approche est acceptable pour se former ou réaliser des démos simples. Cependant, elle va ralentir l'affichage de votre site, elle n'est donc **pas adaptée pour la production**. Lorsque vous serez prêt·e à aller plus loin, supprimez la balise `<script>` et l'attribut `type="text/babel"` que vous venez d'ajouter. Dans la section suivante, nous verrons plutôt comment configurer le préprocesseur JSX afin de convertir automatiquement toutes les balises `<script>`.
```

### Ajouter JSX à votre projet {#add-jsx-to-a-project}

> L'ajout de JSX à votre projet ne nécessite pas d'outils compliqués comme un gestionnaire de dépendances ou un serveur de développement. Cela ressemble plus **à l'ajout d'un préprocesseur CSS**. Le seul outil indispensable est [Node.js](https://nodejs.org/) qui doit être installé sur votre ordinateur.

Reviewer: ```suggestion
L'ajout de JSX à votre projet ne nécessite pas d'outils compliqués comme un _bundler_ ou un serveur de développement. Cela ressemble plus **à l'ajout d'un préprocesseur CSS**. Le seul outil indispensable est [Node.js](https://nodejs.org/fr/), qui doit être installé sur votre ordinateur.
```

Un _bundler_ (comme Webpack) n’est pas un dep manager (comme npm)

Author: Ouais je me retrouve surtout sur le même problème qu'au début sur un bundler intraduisible 😄

> L'ajout de JSX à votre projet ne nécessite pas d'outils compliqués comme un gestionnaire de dépendances ou un serveur de développement. Cela ressemble plus **à l'ajout de préprocesseur CSS**. Le seul outil indispensable est [Node.js](https://nodejs.org/) qui doit être installé sur votre ordinateur.

Reviewer 2: d'un préprocesseur CSS

> Déplacez-vous sur votre projet à l'aide de votre terminal et copiez/collez ces deux commandes :

Reviewer: ```suggestion
Dans un terminal (une ligne de commande), déplacez-vous dans le dossier de votre projet, et copiez-collez ces deux commandes :
```

- Les Windowsiens ne connaissent souvent pas le terme « Terminal », pas utilisé dans leur OS
- Copier-coller (avec un tiret, par un slash)

Author: Merci mais je suis "windowsien"/dev .NET/MCT et fier de l'être 😜 
dans ce cas je dirais plutôt "invite de commande" ou "interface en ligne de commande" meme si on utilise quand même pas mal terminal aussi ^^ 

```suggestion
Dans un terminal (invite de commande), déplacez-vous dans le dossier de votre projet, et copiez-collez ces deux commandes :
```

Reviewer: Oui, « invite de commande » est en effet plus le terme Windows.

> 1. **Étape 1 :** Run `npm init -y` (si ça ne fonctione pas, [voici une correction](https://gist.github.com/gaearon/246f6380610e262f8a648e3e51cad40d))

Reviewer: ```suggestion
1. **Étape 1 :** exécutez `npm init -y` (si ça ne fonctionne pas, [voici un correctif](https://gist.github.com/gaearon/246f6380610e262f8a648e3e51cad40d))
```

> 2. **Étape 2 :** Run `npm install babel-cli@6 babel-preset-react-app@3`

Reviewer: ```suggestion
2. **Étape 2 :** exécutez `npm install babel-cli@6 babel-preset-react-app@3`
```

Dommage qu'ils n'aient pas mis à jour pour Babel 7…

>Astuce
>

> >Nous allons **utiliser npm afin d'installer le préprocesseur JSX** (vous n'en aurez besoin pour rien d'autre). React et le code de l'application peuvent rester sous la forme de balises `<script>` sans modification.

Reviewer: ```suggestion
>Nous **utilisons npm uniquement  pour installer le préprocesseur JSX** (vous n'en aurez besoin pour rien d'autre). React et le code applicatif peuvent rester sous la forme de balises `<script>` sans modification.
```

On vient de l'utiliser, le futur est donc inadéquat.

> Bravo ! Vous venez d'ajouter **la configuration JSX pour la production** à votre projet.

Reviewer: ```suggestion
Bravo ! Vous venez d'ajouter **une configuration JSX de production** à votre projet.
```

### Lancer le préprocesseur JSX {#run-jsx-preprocessor}

> Créez un dossier nommé `src` et lancez la commande suivante dans le terminal :

Reviewer: Bon, vu qu'on a clarifié plus haut, on va laisser « terminal » seul ici

```
npx babel --watch src --out-dir . --presets react-app/prod
```

>Remarque
>

> >`npx` n'est pas une faute de frappe, il s'agit d'un [outil de lancement de package (« package runner tool » dans la langue de Shakespeare, NdT) qui accompagne npm 5.2+](https://medium.com/@maybekatz/introducing-npx-an-npm-package-runner-55f7d4bd282b).

Reviewer: ```suggestion
>`npx` n'est pas une faute de frappe, il s'agit d'un [outil d’exécution de paquet fourni avec npm 5.2+](https://medium.com/@maybekatz/introducing-npx-an-npm-package-runner-55f7d4bd282b) *(”package runner tool”, NdT)*.
```

- L'aparté dans le lien serait visuellement moche
- Les NdT sont intégralement en italiques
- Pour de l'anglais, on utilise les apostrophes anglaises
- Les NdT sont forcément depuis l'anglais, inutile de le préciser à chaque fois

Reviewer: @Nab69 quid de celle-ci ?

> >`npx` n'est pas une faute de frappe, il s'agit d'un [un outil de lancement de package (« package runner tool » dans la langue de Shakespeare, NdT) qui accompagne npm 5.2+](https://medium.com/@maybekatz/introducing-npx-an-npm-package-runner-55f7d4bd282b).

Reviewer 2: il s'agit d'un [un outil

la dessus j'hesite, on garde la repetition du `un` ou on en supprime un des deux ?

Author: j'en supprime un

>

> >Si le message d'erreur « You have mistakenly installed the `babel` package » apparait, vous avez surement dû rater [l'étape précédente](#add-jsx-to-a-project). Dans le même dossier, recommencez l'opération et essayer de nouveau.

Reviewer: ```suggestion
>Si le message d'erreur *”You have mistakenly installed the `babel` package”* apparaît, vous avez peut-être loupé [l'étape précédente](#add-jsx-to-a-project). Dans le même dossier, recommencez l'opération et essayez de nouveau.
```

- Texte anglais entre apostrophes anglaises et en italiques
- “might” est tout sauf « sûr »
- « Rater » peut vouloir dire tant « manquer » (louper) que « foirer », ce qui est plus négatif. On va utiliser le plus détendu et moins ambigu « loupé »

> Cette commande lance un observateur automatisé de votre JSX donc n'attendez pas que cela se termine.

Reviewer: ```suggestion
Cette commande surveille votre JSX en continu, inutile donc d’attendre qu’elle se termine.
```

Allègement

Author: Je voulais garder le sens "watcher" que je trouvais interessant pour la compréhension, mais ça me va.

> Dorénavant, si à partir de **[code de démarrage JSX](https://cdn.rawgit.com/gaearon/c8e112dc74ac44aac4f673f2c39d19d1/raw/09b951c86c1bf1116af741fa4664511f2f179f0a/like_button.js)**, vous créez un fichier nommé `src/like_button.js`, l'observateur va lancer un préprocesseur afin de créer un fichier `like_button.js` avec du code JavaScript utilisable par le navigateur. Ce fichier obtenu sera recréé à chaque modification de votre fichier JSX original.

Reviewer: ```suggestion
Dorénavant, si à partir du **[code de démarrage JSX](https://cdn.rawgit.com/gaearon/c8e112dc74ac44aac4f673f2c39d19d1/raw/09b951c86c1bf1116af741fa4664511f2f179f0a/like_button.js)**, vous créez un fichier nommé `src/like_button.js`, le préprocesseur va produire un fichier traduit `like_button.js`, avec du code JavaScript exploitable par le navigateur. Ce fichier sera mis à jour à chaque modification de votre fichier JSX original.
```

Glissements sémantiques

> Avec ce principe, vous aurez la possibilité d'utiliser des syntaxes modernes de JavaScript comme les classes, sans vous soucier de la compatibilité avec d'anciens navigateurs. L'outil que nous venons de décrire se nomme Babel, vous pouvez en savoir plus sur ce dernier en allant sur [sa documentation](https://babeljs.io/docs/en/babel-cli/).

Reviewer: ```suggestion
Et en bonus, ça vous permet aussi d’utiliser des syntaxes JavaScript modernes comme les classes, sans vous soucier de la compatibilité avec d'anciens navigateurs. L'outil que nous venons d’utiliser se nomme Babel, et vous pouvez en savoir plus sur lui en allant sur [sa documentation](https://babeljs.io/docs/en/babel-cli/).
```

Mais… pourquoi tant de glissements et d'alourdissements ?! Ça imprègne toute ta trad… La fatigue ?

Author: Non pas vraiment de la fatigue, malgré de vrais fautes qui traînaient 😄
C'est plûtot une volonté de garder un peu (pas forcément trop non plus) le coté "documentation écrite" sans trop se laisser aller en mode "oral"

> Si vous vous sentez à l'aise avec les outils de compilation et souhaitez en apprendre plus, la [partie suivante](/docs/create-a-new-react-app.html) décrit les outils les plus populaires et les plus accessibles de la chaine de compilation. Si ce n'est pas le cas, les balises décrites précédemment fonctionneront à merveille !

Reviewer: ```suggestion
Si vous vous sentez à l'aise avec les outils de compilation et souhaitez les utiliser davantage, la [prochaine section](/docs/create-a-new-react-app.html) décrit les boîtes à outils les plus populaires et les plus accessibles. Dans le cas contraire, les balises décrites précédemment fonctionneront à merveille !
```
